src/suggestion.rs

```diff
@@ -50,6 +105,163 @@ impl fmt::Display for Detector {
     }
 }
 
+// For long lines, literal will be trimmed to display in one terminal line
+// Misspelled words that are too long shall also be ellipsized
+pub fn convert_long_statements_to_short(
+    terminal_size: usize,
+    indent: usize,
+    stripped_line: &str,
+    offset: &mut usize,
+    range_word: Range,
+    padding_till_literal_start: usize,
+    marker_size: &mut usize,
+) -> String {
+    use super::*;
+    //
+    // The paddings give some space for the ` {} ...` and extra indentation and formatting:
+    //
+    //|---------------------------------------------------------------------------------------| terminal_size
+    //|-------| padding_till_literal_start = indent (3+line_number_digit_count) + 2 white spaces = 7usize, for this case.
+    //     |------| offset = PADDING_OFFSET; 3 chars for `...` and 2 white spaces more added in the formatting.
+    //
+    //   --> /home/tmhdev/Documents/cargo-spellcheck/src/suggestion.rs:62
+    //    |
+    // 62 |  ... Reasn of food, what's up with pie? There's strawberry pie, apple, pumpkin ...
+    //    |      ^^^^^^
+    //    | - there, Cherie, thither, or tither
+    //    |
+    //    |   Possible spelling mistake found.
+    //
+
+    const PADDING_OFFSET: usize = 5;
+    const TOO_LONG_WORD: usize = 20;
+    const DISPLAYED_LONG_WORD: usize = 4;
+    const PADDING_AROUND_LONG_LINES: usize = 10;
+
+    // We will be using ranges to help doing the fitting:
+    //
+    // |----------------------------------literal_word----------------------------------|
+    // |----------------------|---------misspelled_word---------|-----------------------|
+    // |-----left_context-----|---start_word----|----end_word---|-----right_context-----|
```

Comment:
is this the "word" or rather a `char` index of that line?

Reply:
I change the name because is no longer a literal_word rather than the whole too long line statement.